### PR TITLE
Fixes #15076: Perfect `yii\db\QueryBuilder` buildColumns function

### LINKED ISSUE #15076: yii\db\QueryBuilder buildColumns function
### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.?
| PHP version      | PHP7.1.0
| Operating system | Win10

```
public function buildColumns($columns)
    {
        if (!is_array($columns)) {
            if (strpos($columns, '(') !== false) {
                return $columns;
            } else {
                $columns = preg_split('/\s*,\s*/', $columns, -1, PREG_SPLIT_NO_EMPTY);
            }
        }
        foreach ($columns as $i => $column) {
            if ($column instanceof Expression) {
                $columns[$i] = $column->expression;
            } elseif (strpos($column, '(') === false) {
                $columns[$i] = $this->db->quoteColumnName($column);
            }
        }

        return is_array($columns) ? implode(', ', $columns) : $columns;
    }
```

$columns is a array, so not required use is_array check $columns type, direct use implode(', ', $columns) return.

### PR DESCRIPTION
**Pull request by hiscaler**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #15076

### CODE REVIEW

File: framework/db/QueryBuilder.php [hunk 1]
```diff
@@ -1151,6 +1151,9 @@ public function buildColumns($columns)
             }
 
             $columns = preg_split('/\s*,\s*/', $columns, -1, PREG_SPLIT_NO_EMPTY);
+            if ($columns === false) {
+                return '';
```
Comment (cebe) [inline framework/db/QueryBuilder.php]: what is the error that could occur here? I'm quite sure returning an empty string here will result in an SQL syntax error. It should throw an exception instead.

Comment (hiscaler) [reply framework/db/QueryBuilder.php]: Yes, you are right.here sql is has syntax error if user use invalid fields or string. system will can't continue and hint use check code. Nothing or throw an exception?

Comment (cebe) [reply framework/db/QueryBuilder.php]: throwing a meaningful exception is better than creating an sql error.

Comment (hiscaler) [reply framework/db/QueryBuilder.php]: yes, Thank you. i change it .throw a InvalidParamException exception if `columns` param error.

File: framework/db/QueryBuilder.php [hunk 2]
```diff
@@ -1151,6 +1151,9 @@ public function buildColumns($columns)
             }
 
             $columns = preg_split('/\s*,\s*/', $columns, -1, PREG_SPLIT_NO_EMPTY);
+            if ($columns === false) {
+                throw new InvalidParamException("$columns is not valid columns.");
```
Comment (samdark) [inline framework/db/QueryBuilder.php]: We know it's `false`. Why building a string from a variable?

**Review by samdark — APPROVED** [review]

File: framework/db/QueryBuilder.php (hunk 2 above)
Comment (hiscaler) [reply framework/db/QueryBuilder.php]: I am sorry, is fixed now.

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**